### PR TITLE
Preserve tool call messages in AI chat history to improved LLM context

### PR DESCRIPTION
**Pull request by ravikiranvm**
Fixes OPS-2122.

## Additional Notes
I tested the chat history in the Network tab of browser:
### Before
Screenshot 2025-07-23 at 4 35 28 PM

### After the changes
Screenshot 2025-07-23 at 4 35 28 PM

## Testing Checklist

Check all that apply:

- [x] I tested the feature thoroughly, including edge cases

- [x] I verified all affected areas still work as expected

- [x] Automated tests were added/updated if necessary

- [x] Changes are backwards compatible with any existing data, otherwise a migration script is provided

## Visual Changes (if applicable)

If there are UI/UX changes, include at least one of the following:

- Screenshots
- Loom video
- Preview deployment link

### CODE REVIEW

**Comment by ravikiranvm** [comment]
@rSnapkoOpenOps I've deployed these changes to UX.

File: packages/server/api/src/app/ai/chat/ai-mcp-chat.controller.ts [hunk 1]
```diff
@@ -421,12 +421,10 @@ async function streamMessages(
...
     async onFinish({ response }): Promise<void> {
-      const filteredMessages = removeToolMessages(messages);
-      response.messages.forEach((r) => {
-        filteredMessages.push(getResponseObject(r));
-      });
-
-      await saveChatHistory(chatId, userId, projectId, filteredMessages);
+      await saveChatHistory(chatId, userId, projectId, [
+        ...messages,
+        ...response.messages.map(getResponseObject),
```
[the remark this answers is not in the document]
Comment (rmn-snpk) [reply packages/server/api/src/app/ai/chat/ai-mcp-chat.controller.ts:426]: not relevant
